nuxeo-features/nuxeo-platform-tag-service/nuxeo-platform-tag-core/src/main/java/org/nuxeo/ecm/platform/tag/TagsMigrator.java

```diff
@@ -223,8 +224,16 @@ protected void removeDocument(CoreSession session, String docId) {
     protected void addTags(CoreSession session, String docId, Set<Tag> tags) {
         DocumentModel doc;
         try {
+            if (StringUtils.isEmpty(docId)) {
+                log.warn("TagsRelationsToFacetsMigrator : docId found null in addTags");
+                // ignore null docId
+                return;
+            }
             doc = session.getDocument(new IdRef(docId));
-        } catch (DocumentNotFoundException e) {
+        } catch (DocumentNotFoundException | IllegalArgumentException e) {
+            if (e instanceof IllegalArgumentException) {
```

Comment:
You shouldn't need to catch `IllegalArgumentException` as you're returning before.
If it's still useful, can you use a separate catch to avoid the `instanceof` ?

Reply:
Yes you're already dealing with null now, so `IllegalArgumentException` is not expected anymore.